### DISCUSSION TITLE
How to authenticate and create a release

### DISCUSSION THREAD

**Discussion by sumiticfai196** [microsoft/PowerShellForGitHub#324] [Q&A] [answered]
Hi 

Am pretty new to your module and was exploring it's avenue to be used in Azure Devops to create a release in GitHub Enterprise Repository.

Will it be possible for you to guide me with the steps 

- Connecting to GitHub Enterprise using PAT
- Then pulling out an artifact which has been created
- Creating a release using that artifact

I tried couple of things using the modules but no luck :(

**Comment by HowardWolosky** [comment, marked as answer]
Hi @sumiticfai196,

* Can you include the code of what you've tried so far that hasn't worked and explain how it's failing?
* From where are you "pulling out an artifact which has been created"?

To connect with GitHub Enterprise using a PAT you'll likely need to do the following:
1. [Configure the correct endpoint](https://github.com/microsoft/PowerShellForGitHub#github-enterprise)
2. [Get an API token and call `Set-GitHubAuthentication`](https://github.com/microsoft/PowerShellForGitHub#configuration)

For creating a release, there are [some examples in USAGE.md](https://github.com/microsoft/PowerShellForGitHub/blob/master/USAGE.md#Releases), and you can review the documentation [in the Wiki](https://github.com/microsoft/PowerShellForGitHub/wiki) for those commands to see the full set of what they support. 
You'll need to first [create a new release](https://github.com/microsoft/PowerShellForGitHub/blob/master/USAGE.md#create-a-new-release) and then you can [upload the artifact as an asset](https://github.com/microsoft/PowerShellForGitHub/blob/master/USAGE.md#create-a-release-asset).

**Reply by sumiticfai196** [reply]
Followed following steps 

```powershell
$secureString = ("<redacted>" | ConvertTo-SecureString -AsPlainText -Force)
$cred = New-Object System.Management.Automation.PSCredential "username is ignored", $secureString
Set-GitHubAuthentication -Credential $cred

Set-GitHubConfiguration -ApiHostName "github.sumit.org"
Get-GitHubRepository -Uri "https://github.sumit.org/HP-UTILITY-SYS/KPMETRS/tree/BUG"
New-GitHubRelease -Uri "https://github.sumit.org/HP-UTILITY-SYS/KPMETRS/tree/BUG" -Tag "1.0.0.1" -Name "PowerShell" 
```

Need your suggestion how can i include only a specific folder/artifact for release

**Reply by HowardWolosky** [reply]
> Need your suggestion how can i include only a specific folder/artifact for release

I'm still not sure I follow your question.  Did [this example](https://github.com/microsoft/PowerShellForGitHub/blob/master/USAGE.md#create-a-release-asset) not sufficiently explain how to upload an artifact to be part of the release?

```powershell
New-GitHubReleaseAsset -OwnerName 'HP-UTILITY-SYS' -RepositoryName 'KPMETRS' -Release 123456 -Path 'c:\foo.zip'
```

**Reply by sumiticfai196** [reply]
Thanks @HowardWolosky 
I tried same after posting the question and it really worked for me

**Reply by HowardWolosky** [reply]
Ok, great.  So, your issue is now resolved?
